daemon/api.go

```diff
@@ -770,29 +769,35 @@ func sideloadSnap(c *Command, r *http.Request) Response {
 
 	// form.File is a map of arrays of *FileHeader things
 	// we just allow one (for now at least)
+	var snapBody multipart.File
 out:
 	for _, v := range form.File {
```

Comment:
Can we please be more strict here and require that the form entry be named "snap" instead of simply taking whatever comes and assuming it is the snap?  The client is already doing the right thing in those terms.

In practice:
```
for name, fheaders := range form.File {
        if name != "snap" {
                continue
        }
        for _, fheader := range fheaders {
                snapBody, err := fheader.Open()
                ... etc ...
        }
} 
```

Sample of how curl sends the data:

http://paste.ubuntu.com/16086852/